eleventy-fetch.d.ts

```diff
@@ -0,0 +1,26 @@
+/// <reference types="node" />
+import { Buffer } from 'buffer';
+
+export type FetchType =
+    | "json"
+    | "buffer"
+    | "text";
+
+export type EleventyFetchOptions<TType extends FetchType = "buffer"> = {
+    type?: TType;
+    directory?: string;
+    concurrency?: number;
+    fetchOptions?: RequestInit;
+    dryRun?: boolean;
+    removeUrlQueryParams?: boolean;
+    verbose?: boolean;
+    hashLength?: number;
+    duration?: string;
+    formatUrlForDisplay?: (url: string) => string;
+}
+
+export function EleventyFetch(url: string, options?: EleventyFetchOptions<"buffer">): Buffer;
+export function EleventyFetch<T>(url: string, options: EleventyFetchOptions<"json">): T;
+export function EleventyFetch(url: string, options: EleventyFetchOptions<"text">): string;
```

Comment:
It appears that the return types for these functions should be wrapped in Promise. 
This is consistent with how PQueue's add method is typed, which returns a Promise as seen here: https://github.com/sindresorhus/p-queue/blob/main/source/index.ts#L234